### PR TITLE
GH Action CI Builds: Ninja

### PR DESCRIPTION
**Pull request by ax3l**
Use Ninja for most builds to compile faster than using the GNU Makefile generator.

### CODE REVIEW

**Comment by ax3l** [comment]
## CMake -> GNU Makefiles

```
🍏 macOS / AppleClang (pull_request) Successful in 33m

🐧 CUDA / NVCC 11.0.2 SP (pull_request) Successful in 39m

🐧 CUDA / NVHPC@21.9 NVCC/NVC++ Release [tests] (pull_request) Successful in 79m

🐧 HIP / HIP 3D SP (pull_request) Successful in 30m

🐧 HIP / HIP 2D DP (pull_request) Successful in 36m

🐧 OpenMP / Clang pywarpx (pull_request) Successful in 39m

🐧 Intel / oneAPI ICX SP (pull_request) Successful in 12m

🐧 Intel / oneAPI ICC SP&DP (pull_request) Successful in 30m

🐧 Intel / oneAPI DPC++ SP (pull_request) Successful in 27m 
```

## CMake -> Ninja

```
🍏 macOS / AppleClang (pull_request) Successful in 42m

🐧 CUDA / NVCC 11.0.2 SP (pull_request) Successful in 39m 

🐧 CUDA / NVHPC@21.9 NVCC/NVC++ Release [tests] (pull_request) Successful in 86m 

🐧 HIP / HIP 3D SP (pull_request) Successful in 27m

🐧 HIP / HIP 2D DP (pull_request) Successful in 34m

🐧 OpenMP / Clang pywarpx (pull_request) Successful in 44m

🐧 Intel / oneAPI ICX SP (pull_request) Successful in 13m

🐧 Intel / oneAPI ICC SP&DP (pull_request) Successful in 37m

🐧 Intel / oneAPI DPC++ SP (pull_request) Successful in 34m
```

I kept the tests that are 10% faster in CI (mainly GCC based ones) and did not change the others where I saw slower performance on CI.

Probably depends on many params such as the network FS used in CI.

File: .github/workflows/intel.yml [hunk 1]
```diff
@@ -129,14 +121,11 @@ jobs:
           -DCMAKE_VERBOSE_MAKEFILE=ON  \
           -DWarpX_COMPUTE=SYCL         \
           -DWarpX_EB=ON                \
-          -DWarpX_LIB=OFF              \
+          -DWarpX_LIB=ON               \
```
Comment (ax3l) [inline .github/workflows/intel.yml:127]: Turned on again, since a new OneAPI release came out

File: .github/workflows/ubuntu.yml [hunk 2]
```diff
@@ -10,8 +10,9 @@ jobs:
   build_cxxminimal:
     name: GCC Minimal w/o MPI
     runs-on: ubuntu-20.04
-    env: {CXXFLAGS: "-Werror"}
     if: github.event.pull_request.draft == false
+    env:
+      CXXFLAGS: "-Werror"
```
Comment (ax3l) [inline .github/workflows/ubuntu.yml:15]: Ninja intentionally left out to keep "minimal"

**Comment by ax3l** [comment]
Ready for review & merge now.

**Review by RemiLehe — APPROVED** [review]

**Pull request merged**